src/main/java/org/prebid/server/handler/NotificationEventHandler.java

```diff
@@ -26,109 +32,156 @@
     private static final String PNG_CONTENT_TYPE = "image/png";
 
-    private final AnalyticsReporter analyticsReporter;
-    private final Map<String, TrackingPixel> trackingPixels;
+    private static final String TYPE_PARAMETER = "t";
+    private static final String WIN_TYPE = "w";
+    private static final String IMP_TYPE = "i";
+
+    private static final String BID_ID_PARAMETER = "b";
+    private static final String ACCOUNT_PARAMETER = "a";
+
+    private static final String FORMAT_PARAMETER = "f";
+    private static final String PIXEL_FORMAT = "i";
+    private static final String BLANK_FORMAT = "b";
 
-    private NotificationEventHandler(AnalyticsReporter analyticsReporter, Map<String, TrackingPixel> trackingPixels) {
-        this.analyticsReporter = analyticsReporter;
-        this.trackingPixels = trackingPixels;
+    private static final String ANALYTICS_PARAMETER = "x";
+    private static final String ENABLED_ANALYTICS = "1";
+    private static final String DISABLED_ANALYTICS = "0";
+
+    private final AnalyticsReporter analyticsReporter;
+    private final TrackingPixel trackingPixel;
+    private TimeoutFactory timeoutFactory;
+    private ApplicationSettings applicationSettings;
+
+    public NotificationEventHandler(AnalyticsReporter analyticsReporter, TimeoutFactory timeoutFactory,
+                                    ApplicationSettings applicationSettings) {
+        this.analyticsReporter = Objects.requireNonNull(analyticsReporter);
+        this.timeoutFactory = Objects.requireNonNull(timeoutFactory);
+        this.applicationSettings = Objects.requireNonNull(applicationSettings);
+        this.trackingPixel = TrackingPixel.of(PNG_CONTENT_TYPE, readTrackingPixel(TRACKING_PIXEL_PNG));
     }
 
-    public static NotificationEventHandler create(AnalyticsReporter analyticsReporter) {
-        final Map<String, TrackingPixel> trackingPixels = new HashMap<>();
-        trackingPixels.put(JPG_FORMAT, TrackingPixel.of(JPG_CONTENT_TYPE, readTrackingPixel(TRACKING_PIXEL_JPG)));
-        trackingPixels.put(PNG_FORMAT, TrackingPixel.of(PNG_CONTENT_TYPE, readTrackingPixel(TRACKING_PIXEL_PNG)));
-        return new NotificationEventHandler(Objects.requireNonNull(analyticsReporter), trackingPixels);
+    private static byte[] readTrackingPixel(String path) {
+        try {
+            return ResourceUtil.readByteArrayFromClassPath(path);
+        } catch (IOException e) {
+            throw new IllegalArgumentException(String.format("Failed to load pixel image at %s", path), e);
+        }
     }
 
     @Override
     public void handle(RoutingContext context) {
+        final HttpServerResponse response = context.response();
         final MultiMap queryParameters = context.request().params();
-
-        final NotificationEvent notificationEvent;
         try {
-            notificationEvent = makeNotificationEvent(queryParameters);
+            validateParameters(queryParameters, response);
```

Comment:
Split this method into smth like `validateParametersForBadStatusError(...)` and `validateParametersForUnauthorizedError(...)`. Each method will be executed in separate `try-catch` block with different handling.